// ng-example-library/bundles/ng-example-library.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ng-example-library', ['exports', '@angular/core'], factory) :
    (factory((global['ng-example-library'] = {}),global.ng.core));
}(this, (function (exports,i0) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgExampleLibraryService = (function () {
        function NgExampleLibraryService() {
        }
        /**
         * @return {?}
         */
        NgExampleLibraryService.prototype.getData = /**
         * @return {?}
         */
            function () {
                return 'NgExampleLibraryService';
            };
        NgExampleLibraryService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        NgExampleLibraryService.ctorParameters = function () { return []; };
        /** @nocollapse */ NgExampleLibraryService.ngInjectableDef = i0.defineInjectable({ factory: function NgExampleLibraryService_Factory() { return new NgExampleLibraryService(); }, token: NgExampleLibraryService, providedIn: "root" });
        return NgExampleLibraryService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgExampleLibraryComponent = (function () {
        function NgExampleLibraryComponent() {
        }
        /**
         * @return {?}
         */
        NgExampleLibraryComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        NgExampleLibraryComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'exam-ng-example-library',
                        template: "\n    <p>\n      ng-example-library works!\n    </p>\n  ",
                        styles: []
                    },] },
        ];
        /** @nocollapse */
        NgExampleLibraryComponent.ctorParameters = function () { return []; };
        return NgExampleLibraryComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgExampleLibraryModule = (function () {
        function NgExampleLibraryModule() {
        }
        NgExampleLibraryModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [],
                        declarations: [
                            NgExampleLibraryComponent
                        ],
                        exports: [
                            NgExampleLibraryComponent
                        ]
                    },] },
        ];
        return NgExampleLibraryModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.NgExampleLibraryService = NgExampleLibraryService;
    exports.NgExampleLibraryComponent = NgExampleLibraryComponent;
    exports.NgExampleLibraryModule = NgExampleLibraryModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
