// ng-example-library/fesm2015/ng-example-library.js
import { Injectable, Component, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class NgExampleLibraryService {
    constructor() { }
    /**
     * @return {?}
     */
    getData() {
        return 'NgExampleLibraryService';
    }
}
NgExampleLibraryService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
NgExampleLibraryService.ctorParameters = () => [];
/** @nocollapse */ NgExampleLibraryService.ngInjectableDef = defineInjectable({ factory: function NgExampleLibraryService_Factory() { return new NgExampleLibraryService(); }, token: NgExampleLibraryService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class NgExampleLibraryComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NgExampleLibraryComponent.decorators = [
    { type: Component, args: [{
                selector: 'exam-ng-example-library',
                template: `
    <p>
      ng-example-library works!
    </p>
  `,
                styles: []
            },] },
];
/** @nocollapse */
NgExampleLibraryComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class NgExampleLibraryModule {
}
NgExampleLibraryModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [
                    NgExampleLibraryComponent
                ],
                exports: [
                    NgExampleLibraryComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { NgExampleLibraryService, NgExampleLibraryComponent, NgExampleLibraryModule };
