// ng-example-library/fesm5/ng-example-library.js
import { Injectable, Component, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var NgExampleLibraryService = /** @class */ (function () {
    function NgExampleLibraryService() {
    }
    /**
     * @return {?}
     */
    NgExampleLibraryService.prototype.getData = /**
     * @return {?}
     */
    function () {
        return 'NgExampleLibraryService';
    };
    NgExampleLibraryService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgExampleLibraryService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgExampleLibraryService.ngInjectableDef = defineInjectable({ factory: function NgExampleLibraryService_Factory() { return new NgExampleLibraryService(); }, token: NgExampleLibraryService, providedIn: "root" });
    return NgExampleLibraryService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var NgExampleLibraryComponent = /** @class */ (function () {
    function NgExampleLibraryComponent() {
    }
    /**
     * @return {?}
     */
    NgExampleLibraryComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    NgExampleLibraryComponent.decorators = [
        { type: Component, args: [{
                    selector: 'exam-ng-example-library',
                    template: "\n    <p>\n      ng-example-library works!\n    </p>\n  ",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    NgExampleLibraryComponent.ctorParameters = function () { return []; };
    return NgExampleLibraryComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var NgExampleLibraryModule = /** @class */ (function () {
    function NgExampleLibraryModule() {
    }
    NgExampleLibraryModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [
                        NgExampleLibraryComponent
                    ],
                    exports: [
                        NgExampleLibraryComponent
                    ]
                },] },
    ];
    return NgExampleLibraryModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { NgExampleLibraryService, NgExampleLibraryComponent, NgExampleLibraryModule };
